src/Network/Request.php

```diff
@@ -1405,16 +1413,14 @@ public function param($name)
      * @param string|null $callback A decoding callback that will convert the string data to another
      *     representation. Leave empty to access the raw input data. You can also
      *     supply additional parameters for the decoding callback using var args, see above.
+     * @param array ...$args The additional arguments
      * @return string The decoded/processed request data.
      */
-    public function input($callback = null)
+    public function input($callback = null, ...$args)
     {
-        $input = $this->_readInput();
-        $args = func_get_args();
-        if (!empty($args)) {
-            $callback = array_shift($args);
+        $input = $this->stream->getContents();
+        if ($callback) {
             array_unshift($args, $input);
-
             return call_user_func_array($callback, $args);
```

Comment:
I tried that but the tests oddly failed with  `PHP Fatal error:  Call to undefined function Cake\Utility\Xml::build() in /Users/markstory/Sites/cake3/src/Network/Request.php on line 1424`